### DISCUSSION TITLE
Acces request variables in field resolver

### DISCUSSION THREAD

**Discussion by vytautas-pranskunas-** [MichalLytek/type-graphql#800] [Q&A] [answered]
Hi,

I am migrating to type-graphql and previously i havd this field resolver:

```
hasNewChatMessages: {
            type: GraphQLBoolean,
            resolve: (source: Inquiry, context, args, info) => {
                return source.messages.findIndex(m => m.userId.toHexString() !== (info.variableValues as ListItemsFilter).userId && !m.seen) !== -1;
            },
        },
```

cannot find how to access request args (**info.variableValues**) in field resolver.

```
@FieldResolver(() => Boolean)
    hasNewChatMessages(@Root() order: Order) {
        return order.messages.findIndex(m => m.userId.toHexString() !== ?????????? && !m.seen) !== -1;
    }
```

**Comment by MichalLytek** [comment, marked as answer]
`@Info()`

**Reply by vytautas-pranskunas-** [reply]
Can you share a link to documentation?

**Reply by MichalLytek** [reply]
There's no API reference as it's super straightforward:
```ts
hasNewChatMessages(@Root() order: Order, @Info() info: GraphQLResolveInfo) {
  return order.messages.findIndex(m => m.userId.toHexString() !== (info.variableValues as ListItemsFilter).userId && !m.seen) !== -1;
}
```

**Reply by chinanderm** [reply]
Thanks for sharing this insight @MichalLytek! It's too bad it isn't in the documentation as it seems like a useful decorator that users should know about.

**Comment by vytautas-pranskunas-** [comment]
It is when you know it 😉 it is not so obvious that you can reference
additional paramos and more not obvious their types.

Thanks

**Reply by MichalLytek** [reply]
Types are the same as you would write your resolvers in plain apollo style but just with TS in it:
```ts
resolve: (source: Inquiry, context: MyContext, args: SomeArgs, info: GraphQLResolveInfo) 
```

**Reply by vytautas-pranskunas-** [reply]
Thanks,

got ne more issue - when trying to query endpoint getting this error: "Unknown type \"Int\". Did you mean \"ID\"?", on $currentPage: Int!. When changing to Float - works. However in our docs saying that Int is supported: https://typegraphql.com/docs/scalars.html

**Reply by MichalLytek** [reply]
`@Field(type => Int)`
